kge/job/eval.py

```diff
@@ -83,20 +87,36 @@ def run(self) -> dict:
         """ Compute evaluation metrics, output results to trace file """
         raise NotImplementedError
 
-    def resume(self, checkpoint_file=None):
-        """Load model state from last or specified checkpoint."""
-        # load model
-        from kge.job import TrainingJob
-
-        training_job = TrainingJob.create(self.config, self.dataset)
-        training_job.resume(checkpoint_file)
-        self.model = training_job.model
-        self.epoch = training_job.epoch
-        self.resumed_from_job_id = training_job.resumed_from_job_id
+    def load(self, checkpoint, model):
+        if checkpoint["type"] not in ["train", "package"]:
+            raise ValueError("Can only evaluate train and package checkpoints.")
+        self.resumed_from_job_id = checkpoint.get("job_id")
+        self.epoch = checkpoint["epoch"]
         self.trace(
-            event="job_resumed", epoch=self.epoch, checkpoint_file=checkpoint_file
+            event="job_resumed", epoch=self.epoch, checkpoint_file=checkpoint["file"]
         )
 
+    @classmethod
+    def create_from(
+        cls,
+        checkpoint: Dict,
+        overwrite_config: Config = None,
+        dataset: Dataset = None,
+        parent_job=None,
+        eval_split: str = "valid",
```

Comment:
Drop this argument. It's not always used. If the split should be set, if should be done in the overwrite_config.

Reply:
I added this argument, so that the user can easily evaluate without a config. With that he can just call EntityRankingJob.create_from(checkpoint_file, eval_split="test") to evaluate on test.
If we remove it, he has to create a complete empty config file with only the eval split specified.

Reply:
documented overwrite priority